sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/implementation/serializer/DeserializationHelpers.java

```diff
@@ -0,0 +1,23 @@
+package com.azure.digitaltwins.core.implementation.serializer;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class DeserializationHelpers {
+
+    /**
+     * Converts the payload object into a generic type.
+     * There are two different paths we will have to take based on the type T
+     * In case of a String, we need to write the value of the payload as a String
+     * In case of any other type that the user decides to deserialize the payload, we will use mapper.convertValue to perform the casting.
```

Comment:
nit: maybe say "...perform the conversion.", or "...perform the deserialization.". The term "casting" here might be misleading.